badge-maker/package.json

```
@@ -2,7 +2,12 @@
   "name": "badge-maker",
   "version": "5.0.0",
   "type": "module",
-  "exports": "./lib/index.js",
+  "exports": {
+    ".": {
+      "import": "./lib/index.js",
+      "types": "./index.d.ts"
```

This might have a drastic effect but the typescript docs [here](https://www.typescriptlang.org/docs/handbook/modules/reference.html#package-exports:~:text=be%20specifying%20conditions.-,Does%20the%20first%20condition%20%22types%22%20match%20this%20request%3F%20Yes.,-Does%20the%20path) show types first in the examples and explicitly state in the logic they are looking for it as a first property of a subpath field.
```suggestion
      "types": "./index.d.ts",
      "import": "./lib/index.js"
```

---

As I understand it, this only applies to objects with several `types` entries. But it shouldn't make any difference. See your example with multiple conditions:

```json
{
  "name": "pkg",
  "exports": {
    "./subpath": {
      "types@>=5.2": "./ts5.2/subpath/index.d.ts",
      "types@>=4.6": "./ts4.6/subpath/index.d.ts",
      "types": "./tsold/subpath/index.d.ts",
      "default": "./dist/subpath/index.js"
    }
  }
}
```

---

That might be that, regardless it seems to work as is, so i will close this conversation